### DISCUSSION TITLE
CDN and SSG (fallback and revalidate)

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#20236] [Help]
Trying to better understand the expected behavior with SSG when a CDN is used.

For our use case, 98% of the traffic is handled by the CDN.  The only times we hit Next.js server is if we have a route not cached yet (SSG with fallback=true) or we have flushed the cache for a URL where the backend data has changed and we want to see the page update (SSG with revalidate).

My question is with both of these settings, from the documentation appear to return a initial page before the finalized page is completed.  

For fallback, a fallback version of the page can be returned like a loader, then not sure if the page dynamically updates once Next.js has completed the process.  But not sure how that works with a CDN.  Does the initial page get returned to the browser where the isFallback executes and shows the loader then that page makes a call to Next.js to return the json to update the page when it is complete.  Then the question is, as new visitors hit that page that is cached in the CDN, are they just hitting the loader page instead of the finalized page that is cached in Next.js?

For revalidate, similar type of question.  If someone accesses a route for a page and it returns v1 of the page, then the CDN caches that v1 for the route.  If data in the backed is updated, we flush that route in the CDN so the request flow back to the Next.js server.  This is the second hit Next.js gets for this route (since the CDN was caching any previous outside of the first hit), so Next.js starts the revalidate process by returning the v1 static version of the page while it builds v2.  But the question is, is the CDN now going to cache that v1 version again so that v2 isn't seen unless the cache is flushed again?

Just trying to understand the CDN and SSG lifecycle for fallback and revalidate.

**Comment by leerob** [comment]
This might help explain! https://youtu.be/eMwTEo6AjDc?t=571
